### DISCUSSION TITLE
Any reason for the following behavior?

### DISCUSSION THREAD

**Discussion by ashmaroli** [ruby/psych#581] [Q&A] [answered]
Given the following Ruby script:
```ruby
require 'yaml'
puts YAML::VERSION

text = <<~TXT
  items:
    tags:
      - 0001
      - 0002
      - 0003
      - 0004
      - 0005
      - 0006
      - 0007
      - 0008
      - 0009
TXT

puts YAML.dump YAML.load(text)
```
when run outputs the following:
```
3.3.0
---
items:
  tags:
  - 1
  - 2
  - 3
  - 4
  - 5
  - 6
  - 7
  - '0008'
  - '0009'
```

Any reason why there's a special behavior for numbers `0-7` but not for `8` and `9`?

**Comment by perlpunk** [comment, marked as answer]
ruby psych implements YAML 1.1. https://perlpunk.github.io/yaml-test-schema/schemas.html
`0001` - `0007` are read as octal numbers. `0008` and `0009` can't be octal, so they are resolved as a string. When dumping again, I guess they are quoted to make clear that they are strings, but the quotes wouldn't be strictly required.
